File: BackEnd/ExamManagement/DPD.HR.Business.Dto/Dto/NavigationDto.cs
namespace DPD.HumanResources.Dtos.Dto;

public class NavigationDto
{
    public Guid Id {get;set;}
    public string MenuName {get;set;}
    public Guid? IdParent {get;set;}
    public int? Status {get;set;}
    public DateTime? CreatedDate {get;set;}
    public string? Path {get;set;}
    public string? IconLink {get;set;}
    public string? MenuCode {get;set;}
    public int? Sort {get;set;}
}

File: FrontEnd/src/models/navigationDTO.ts
import React from "react";

export interface NavigationDTO {
  navigation: Navigation;
  navigationsChild: Navigation[] | null;
}

export interface Navigation {
  id: string;
  menuName: string;
  idParent: string | null;
  status: number | null;
  createdDate: Date | null;
  path: string;
  iconLink: string;
  menuCode: string;
  sort: number | null;
  icon: React.JSX.Element[];
  children: Navigation[] | null;
}
